spec/unit/puppet-lint/checkplugin_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+
+class DummyCheckPlugin < PuppetLint::CheckPlugin
+  def check
+    # Since we're calling `tokens` from a `check` method, we should get our own Array object.
+    # If we add an extra token to it, PuppetLint::Data.tokens should remain unchanged.
+    tokens << :extra_token
```

Comment:
When returning a frozen array it will raise a `FrozenError`. Perhaps we should catch that and raise a deprecation warning?